src/annotator/plugin/bucket-bar.js

```diff
@@ -69,133 +97,181 @@ export default class BucketBar extends Delegator {
+     * within a given bucket (collection of Anchors).
+     *
+     * @param {Bucket} bucket
+     * @param {boolean} toggle
+     */
+    function focusHighlights(bucket, toggle) {
+      bucket.anchors.forEach(anchor => {
+        anchor.highlights?.forEach(highlight => {
+          if (toggle) {
+            highlight.classList.add('hypothesis-highlight-focused');
+          } else {
+            highlight.classList.remove('hypothesis-highlight-focused');
+          }
+        });
+      });
+    }
+
+    // The following affordances attempt to reuse existing DOM elements
+    // when reconstructing bucket "tabs" to cut down on the number of elements
+    // created and added to the DOM
+
+    // Only leave as many "tab" elements attached to the DOM as there are
+    // buckets
+    this.tabs.slice(this.buckets.length).forEach(tabEl => {
+      this.element.removeChild(tabEl);
+    });
+    // And cut the "tabs" collection down to the size of buckets, too
+    /** @type {HTMLElement[]} */
     this.tabs = this.tabs.slice(0, this.buckets.length);
 
-    // Create any new tabs if needed.
-    $.each(this.buckets.slice(this.tabs.length), () => {
-      const div = $('<div/>').appendTo(this.element);
-
-      this.tabs.push(div[0]);
-
-      div
-        .addClass('annotator-bucket-indicator')
-
-        // Focus corresponding highlights bucket when mouse is hovered
-        // TODO: This should use event delegation on the container.
-        .on('mousemove', event => {
-          const bucketIndex = this.tabs.index(event.currentTarget);
-          for (let anchor of this.annotator.anchors) {
-            const toggle = this.buckets[bucketIndex].includes(anchor);
-            $(anchor.highlights).toggleClass(
-              'hypothesis-highlight-focused',
-              toggle
-            );
-          }
-        })
+    // If the number of "tabs" currently in the DOM is too small (fewer than
+    // buckets), fill that gap by creating new elements (and adding event
+    // listeners to them)
+    this.buckets.slice(this.tabs.length).forEach(() => {
+      const tabEl = document.createElement('div');
+      this.tabs.push(tabEl);
+
+      // Note that these elements are reused as buckets change, meaning that
+      // any given tab element will correspond to a different bucket over time.
+      // However, we know that we have one "tab" per bucket, in order,
+      // so we can look up the correct bucket for a tab at event time.
+
+      // Focus and unfocus highlights on mouse events
+      tabEl.addEventListener('mousemove', () => {
+        const index = this.tabs.indexOf(tabEl);
+        if (
+          index > 0 &&
+          this.buckets[index] &&
+          !this.isNavigationBucket(index)
+        ) {
+          focusHighlights(this.buckets[index], true);
+        }
+      });
+
+      tabEl.addEventListener('mouseout', () => {
+        const index = this.tabs.indexOf(tabEl);
+        if (index > 0 && this.buckets[index]) {
```

Comment:
I can add that, sure. Because the navigation buckets are excluded in the `mousemove` callback, un-highlighting indiscriminately here isn't problematic as the anchor-highlights in question weren't highlighted/focused in the first place, but for neatness sake, I'm glad to add this.